components/admin_console/team_channel_settings/channel/details/channel_details.tsx

```diff
@@ -31,39 +31,39 @@ import SaveChangesPanel from '../../save_changes_panel';
 import {ChannelModes} from './channel_modes';
 import {ChannelGroups} from './channel_groups';
 import {ChannelProfile} from './channel_profile';
-import ChannelModeration from './channel_moderation';
 import ChannelMembers from './channel_members';
+import ChannelModeration from './channel_moderation';
 
-interface ChannelDetailsProps {
+export interface ChannelDetailsProps {
     channelID: string;
     channel: Channel;
     team: Partial<Team>;
     groups: Group[];
     totalGroups: number;
-    channelPermissions?: Array<ChannelPermissions>;
     allGroups: Dictionary<Group>;
+    channelPermissions?: Array<ChannelPermissions>;
     teamScheme?: Scheme;
     guestAccountsEnabled: boolean;
-    isDisabled: boolean;
+    isDisabled?: boolean;
     actions: {
-        getGroups: (channelID: string, q?: string, page?: number, perPage?: number) => Promise<Partial<Group>[]>;
-        linkGroupSyncable: (groupID: string, syncableID: string, syncableType: string, patch: Partial<SyncablePatch>) => ActionFunc|ActionResult;
-        unlinkGroupSyncable: (groupID: string, syncableID: string, syncableType: string) => ActionFunc;
+        getGroups: (channelID: string, q?: string, page?: number, perPage?: number, filterAllowReference?: boolean) => any;
+        linkGroupSyncable: (groupID: string, syncableID: string, syncableType: SyncableType, patch: SyncablePatch) => ActionFunc|ActionResult;
+        unlinkGroupSyncable: (groupID: string, syncableID: string, syncableType: SyncableType) => ActionFunc;
         membersMinusGroupMembers: (channelID: string, groupIDs: Array<string>, page?: number, perPage?: number) => ActionFunc|ActionResult;
-        setNavigationBlocked: (blocked: boolean) => any;
+        setNavigationBlocked: (blocked: boolean) => {type: 'SET_NAVIGATION_BLOCKED', blocked: boolean};
         getChannel: (channelId: string) => ActionFunc;
-        getTeam: (teamId: string) => Promise<ActionResult>;
-        getChannelModerations: (channelId: string) => Promise<Array<ChannelPermissions>>;
+        getTeam: (teamId: string) => any;
+        getChannelModerations: (channelId: string) => any;
         patchChannel: (channelId: string, patch: Channel) => ActionFunc;
-        updateChannelPrivacy: (channelId: string, privacy: string) => Promise<ActionResult>;
-        patchGroupSyncable: (groupID: string, syncableID: string, syncableType: string, patch: Partial<SyncablePatch>) => ActionFunc;
+        updateChannelPrivacy: (channelId: string, privacy: string) => any;
```

Comment:
I think this returns `{data: Channel, error: ServerError}`

Reply:
You may talk about `patchChannelModerations`, aren't you? Same as above, there is a diff with redux function declaration that prevent using it, that's why I put `any`. Any better idea welcomed!

Reply:
Solved with you previous comment